File: datasets/jl-keep-code_davinci_001_temp_0.2-0/HumanEval_138_is_equal_to_sum_even.jl
"""Evaluate whether the given number n can be written as the sum of exactly 4 positive even numbers
    Example
    is_equal_to_sum_even(4) == False
    is_equal_to_sum_even(6) == False
    is_equal_to_sum_even(8) == True
    """
function is_equal_to_sum_even(n::Int64)::Bool 
    if n%2 != 0
        return false
    end
    for i in 1:n
        for j in 1:n
            for k in 1:n
                for l in 1:n
                    if i+j+k+l == n && i%2 == 0 && j%2 == 0 && k%2 == 0 && l%2 == 0
                        return true
                    end
                end
            end
        end
    end
    return false
end
is_equal_to_sum_even(4)
is_equal_to_sum_even(6)
is_equal_to_sum_even(8)

using Test

@testset begin

candidate = is_equal_to_sum_even;
	@test(candidate(4) == false)
	@test(candidate(6) == false)
	@test(candidate(8) == true)
	@test(candidate(10) == true)
	@test(candidate(11) == false)
	@test(candidate(12) == true)
	@test(candidate(13) == false)
	@test(candidate(16) == true)
end

File: datasets/ts-keep-code_davinci_001_temp_0.2/HumanEval_138_is_equal_to_sum_even.ts
//Evaluate whether the given number n can be written as the sum of exactly 4 positive even numbers
// Example
// is_equal_to_sum_even(4) == False
// is_equal_to_sum_even(6) == False
// is_equal_to_sum_even(8) == True
function is_equal_to_sum_even(n: number): boolean {
    let sum = 0;
    for (let i = 0; i < n; i++) {
        sum += i * 2;
    }
    return sum === n;
}


declare var require: any;
const assert = require('node:assert');


function test() {
  let candidate = is_equal_to_sum_even;
  assert.deepEqual(candidate(4),false);
  assert.deepEqual(candidate(6),false);
  assert.deepEqual(candidate(8),true);
  assert.deepEqual(candidate(10),true);
  assert.deepEqual(candidate(11),false);
  assert.deepEqual(candidate(12),true);
  assert.deepEqual(candidate(13),false);
  assert.deepEqual(candidate(16),true);
}

test();